ebean-datasource/src/main/java/io/ebean/datasource/pool/PooledConnection.java

```diff
@@ -122,10 +150,17 @@ final class PooledConnection extends ConnectionDelegator {
     this.name = pool.name() + uniqueId;
     this.originalSchema = pool.schema();
     this.originalCatalog = pool.catalog();
-    this.initialisedSchema = originalSchema != null;
-    this.initialisedCatalog = originalCatalog != null;
-    this.currentSchema = originalSchema != null ? originalSchema : "@default";
-    this.currentCatalog = originalCatalog != null ? originalCatalog : "@default";
+    if (originalSchema != null) {
+      schemaState = SCHEMA_CATALOG_KNOWN;
+      this.cacheKeySchema = originalSchema;
+      // if schema & catalog is defined, we can be sure, that connection is initialized properly
+      assert originalSchema.equals(connection.getSchema()) : "Connection is in the wrong schema: " + connection.getSchema() + ", expected: " + originalSchema;
+    }
+    if (originalCatalog != null) {
+      catalogState = SCHEMA_CATALOG_KNOWN;
+      this.cacheKeyCatalog = originalCatalog;
+      assert originalCatalog.equals(connection.getCatalog()) : "Connection is in the wrong catalog: " + connection.getCatalog() + ", expected: " + originalCatalog;
```

Comment:
I have added some asserts here. Normally, they should apply, but I'm unsure, if we should keep them in production, as some tests are failing, but it helped me to discover the servere bug:

for some DBMS you will not get back the same string; e.g. lower/upper, DB2 may even pad with spaces. (did not check this in detail, I only remember, that it is stored in the catalog as `SCHEMA1 ` with some additional spaces at the end)

postgres is not roundtrip aware. If you set schema that does not exist, you will get `null`
```java
pgConnection.setSchema("public");
pgConnection.getSchema(); // returns public
pgConnection.setSchema("doesnotexist");
pgConnection.getSchema(); // returns null
```

Reply:
I think we remove the asserts as they have potential for side effects.